### PR TITLE
ENG-1088 Implement Email Configuration feature

### PR DESCRIPTION
**Pull request by raxkaynan**
* Adds the Email Configuration sidebar menu item and page.
* Adds the SMTP Server Configuration page.
* Adds the Sender Management page.
* Implements the add, edit, and delete email sender functionality.
* Adds reusable test utility functions.
* Improve accessibility of certain common components.

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **raxkaynan**

If they are not already assigned, you can assign the PR to them by writing `/assign @raxkaynan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: test/test/testUtils.js
Comment (gidesan) [inline test/test/testUtils.js:146]: what about having
- a separate file for RTL utilities
- a single wrapper including all our providers so devs don't have to think about what to use every time?

**Review by gidesan — CHANGES_REQUESTED** [review]
the PR looks very good, I only have a concert about testUtils :)

File: test/test/testUtils.js
Comment (raxkaynan) [reply test/test/testUtils.js:146]: @gidesan I agree with the separate RTL utils file.

However, I'm not sure about the single wrapper that includes all providers. If the dev writes a test for a component that they know only uses a router, then they should use the render function with the router provider. I believe it adds a very small overhead as well if all providers are included always every render.

Comment (gidesan) [reply test/test/testUtils.js:146]: But the real app uses all those providers. That means the UI component tests would be a bit less "unit" and a bit more "integration", which is not bad for me.

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1023#pullrequestreview-605000000" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
